### DISCUSSION TITLE
Multi-Site installation with only subfolders and redirects from non-subfolder hostname

### DISCUSSION THREAD

**Discussion by jannisborgers** [craftcms/cms#10683] [Q&A]
I’m maintaining an installation that has 4 sites:

1. `de.example.com/a` (disabled, former primary site, `default` site handle)
2. `en.example.com/a` (disabled)
3. `de.example.com/b` (primary site)
4. `en.example.com/b`

I have set up the baseUrls for each of the sites in the control panel using environment variables.

However, the site is still accessible via requests to `de.example.com` and `en.example.com`, showing site 3 (`de.example.com/b`). This should be avoided.

I understand that this is because the hostname is pointing to the `/web` folder of the project. I found [this approach](https://craftcms.stackexchange.com/questions/23752/craft-3-multi-site-with-no-default-site) described by @brandonkelly a couple of years ago. How I understand it for my example: Requests to web/index.php have to be set to redirect to `/b` so that there’s no way to visit URLs without a path segment.

I don't know how to do step 3 (locally with Nitro) for testing the solution. Without write privileges for server configuration with the client’s hoster, I would have to communicate with them via a ticketing system, so testing takes a long time. Sooo, any solution that is ready-to-go would be really helpful. 😬

**Comment by timkelty** [comment]
@jannisborgers what do you want to happen with a request to the root de.example.com? 404?

**Reply by jannisborgers** [reply]
@timkelty That should redirect to de.example.com/b, which will be changed every year. So next year this time, I want to change it to point to de.example.com/c, which will then have been created.

**Reply by timkelty** [reply]
Nitro 2 doesn't allow you to add any custom web server/nginx rules (Nitro 3 will), so you'd have to do the logic in PHP. That said, your use case is quite a bit simpler than the link you shared.

In this case, you should be able to do something as simple as this in your `web/index.php`:

```
// Load Craft (but don't run)
/** @var \craft\web\Application $app */
$app = require CRAFT_VENDOR_PATH.'/craftcms/cms/bootstrap/web.php';

if ($app->getRequest()->getFullPath() === "") {
    $app->getResponse()->redirect(getenv('PRIMARY_SITE_URL'))->send();
}

$app->run();
```

**Reply by jannisborgers** [reply]
Thank you @timkelty! Sorry for the delayed response. 

We managed to get the site(s) to work as expected by writing a custom redirect module that redirects to the corresponding „subfolder“ URL if a request URL without a subfolder segment is detected. 

We could have gone with the subfolder inside `/web`, but the client‘s IT hadn’t given us write permissions to `.htaccess` at that point which made that a bit difficult… 😬
